File: backend/src/Hotel.Orbital.Core/Models/SpecialOfferDto.cs
using System.ComponentModel.DataAnnotations;
using Entities.Enums;

namespace Core.Models;

/// <summary>
/// Dto спецпредложения
/// </summary>
public class SpecialOfferDto
{
    /// <summary>
    /// Идентфиикатор
    /// </summary>
    [Required]
    public Guid Id { get; set; }
    
    /// <summary>
    /// Заголовок
    /// </summary>
    [Required]
    public Dictionary<Language, string> Titles { get; set; }
    
    /// <summary>
    /// Краткое описание
    /// </summary>
    [Required]
    public Dictionary<Language, string> ShortDescriptions { get; set; }

    /// <summary>
    /// Описание
    /// </summary>
    [Required]
    public Dictionary<Language, string> Descriptions { get; set; }

    /// <summary>
    /// Заметки
    /// </summary>
    [Required]
    public Dictionary<Language, string?> Notes { get; set; }
    
    /// <summary>
    /// Номера талефона №1
    /// </summary>
    [Phone]
    public string? PhoneNumber1 { get; set; }
    
    /// <summary>
    /// Номера талефона №2
    /// </summary>
    [Phone]
    public string? PhoneNumber2 { get; set; }
    
    /// <summary>
    /// Идентификатор обложки
    /// </summary>
    [Required]
    public ImageDto Cover { get; set; }

    /// <summary>
    /// Идентификаторы изображений
    /// </summary>
    public List<ImageDto> Images { get; set; }
}

File: admin/src/features/special-offers/api/dto/special-offer-dto.ts
import { ImageDto, fromImageDto } from "@/features/images";
import { LangRecordDto, fromLangRecordDto } from "@/features/misc";
import { SpecialOffer } from "../../types/special-offer";

export interface SpecialOfferDto {
  id: string;
  titles: LangRecordDto<string>;
  shortDescriptions: LangRecordDto<string>;
  descriptions: LangRecordDto<string>;
  notes: LangRecordDto<string>;
  phoneNumber1: string;
  phoneNumber2: string;
  cover: ImageDto;
  images: ImageDto[];
}

export const fromSpecialOfferDto = (dto: SpecialOfferDto): SpecialOffer => ({
  id: dto.id,
  titles: fromLangRecordDto(dto.titles),
  shortDescriptions: fromLangRecordDto(dto.shortDescriptions),
  descriptions: fromLangRecordDto(dto.descriptions),
  notes: fromLangRecordDto(dto.notes),
  primaryPhone: dto.phoneNumber1,
  secondaryPhone: dto.phoneNumber2,
  cover: fromImageDto(dto.cover),
  images: dto.images.map(fromImageDto),
});
